src/main/js/helpers/region.js

```diff
@@ -197,23 +197,43 @@ const translateRegion = (scale, config, regionGroupSVG) =>
             return getRegionHeight(d3.select(this), d, scale, config);
         });
 /**
- * Decides if regions needs to be hidden based on count of displayed targets in graph
+ * Checks if only 1 content item is present in the graph
  *
  * @private
- * @param {Array} regionList - List of regions to be shown within graph
+ *
  * @param {Array} graphTargets - List of all the items in the Graph
- * @returns {boolean} true if displayed targets are more than 1, false otherwise
+ * @returns {boolean} true if displayed targets is equal to 1, false otherwise
  */
-const shouldHideAllRegions = (regionList, graphTargets) =>
-    utils.notEmpty(regionList) && graphTargets.length > 1;
+const isSingleTargetDisplayed = (graphTargets) => graphTargets.length === 1;
+
 /**
- * Checks if only 1 content item is present in the graph
+ * Check all the regions within a graph is same or not
  *
  * @private
- * @param {Array} graphTargets - List of all the items in the Graph
- * @returns {boolean} true if displayed targets is equal to 1, false otherwise
+ * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
+ * @returns {boolean} - returns true is regions are same else false
  */
-const isSingleTargetDisplayed = (graphTargets) => graphTargets.length === 1;
+const areRegionSame = (canvasSVG) => {
+    const regionData = [];
+    const regions = canvasSVG.selectAll(`.${styles.region}`).data();
+    regions !== undefined
+        ? regions.forEach((element) => {
+              regionData.push(element);
+          })
+        : regionData.push();
+    const compare = regionData[0];
+    const isSame = regionData.every((element) => {
```

Comment:
The reason I asked you to use `some` instead of `every` is that you wont get immediate feedback. 
If you use `every` then even if you have a non-matching item, the function would just continue to put every item through that check. 
`Some` will fail fast.

Reply:
yes right